File: app/src/main/java/siergo_o/onlinernews/model/RssService.java
package siergo_o.onlinernews.model;

import retrofit2.Retrofit;
import retrofit2.converter.simplexml.SimpleXmlConverterFactory;

public class RssService {

    private Retrofit mRetrofit;
    private String urlLink;

    public RssService(String urlLink){
        mRetrofit = new Retrofit.Builder().baseUrl(urlLink)
                .addConverterFactory(SimpleXmlConverterFactory.create()).build();
    }

//    public RssService(){
//        mRetrofit = new Retrofit.Builder().baseUrl(urlLink)
//                .addConverterFactory(SimpleXmlConverterFactory.create()).build();
//    }

    public static RssService getInstance(String urlLink){

        return new RssService(urlLink);
    }

    public OnlinerAPI getOnlinerAPI(){
        return mRetrofit.create(OnlinerAPI.class);
    }

    public Retrofit getmRetrofit() {
        return mRetrofit;
    }

    public String getUrlLink() {
        return urlLink;
    }

    public void setUrlLink(String urlLink) {

        this.urlLink = urlLink;
    }

}

File: app/src/main/java/siergo_o/onlinernews/domain/news/model/RssService.kt
package siergo_o.onlinernews.domain.news.model

import retrofit2.Retrofit
import retrofit2.converter.simplexml.SimpleXmlConverterFactory
import siergo_o.onlinernews.data.rest.OnlinerApi

class RssService(private val urlLink: String) {
    private val mRetrofit: Retrofit = Retrofit.Builder().baseUrl(urlLink)
            .addConverterFactory(SimpleXmlConverterFactory.create()).build()
    val onlinerApi: OnlinerApi = mRetrofit.create(OnlinerApi::class.java) // TODO Improve retrofit usage

    companion object {
        fun getInstance(urlLink: String): RssService {
            return RssService(urlLink)
        }
    }
}